### PR TITLE
add information about packaging crate

### PR DESCRIPTION
**Pull request by jonaspleyer**
This is a minor addition. Usually when working with the `build crate -l` command, it is enough to apply it directly to the crate root.
However, when using cargo features such as inheritance of workspace settings (ie. `version.workspace = true`) it is necessary to package the crate first with `cargo package`.

I am aware that the wording "package" should already make this clear. However, I stumbled across this problem personally and could not figure out the solution. Please correct me if I have overlooked any other documentation where this makes more sense.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -207,6 +207,9 @@ cargo run -- build world
 # Builds a local package you have at <SOURCE> and adds it to the database.
 # The package does not have to be on crates.io.
 # The package must be on the local filesystem, git urls are not allowed.
+# Usually this command can be applied directly to a crate root
+# In certain scenarios it might be necessary to first package the respective
```
Comment (syphar) [inline README.md:211]: Could you add some examples here which cases might need `cargo package`? 

Also, how the `cargo package` command would relate to the local build command itself?

**Review by syphar — COMMENTED** [review]
Thank you for the improvement! This really helps!

File: README.md (hunk 1 above)
Comment (Nemo157) [reply README.md:211]: I wonder if we could have building a local crate package it then build from the package directly. Might need some minor rustwide changes to support.

**Comment by jonaspleyer** [comment]
Hey everyone. Sorry for taking so long. I have created a sample repo which showcases how workspaces packages cannot be build by using the commands as explained in the README.md file.

# Example
https://github.com/jonaspleyer/docs_rs_workspace_package

## Structure
```bash
$ tree
.
├── Cargo.toml
├── my_lib
│   ├── Cargo.toml
│   └── src
│       └── lib.rs
└── README.md

3 directories, 4 files
```
The actual contents of `my_lib` do not matter, only the two configuration files.
```
$ cat Cargo.toml
[workspace]
members = [
        "my_lib",
]

[workspace.package]
version = "0.1.0"

$ cat my_lib/Cargo.toml
[package]
name = "my_lib"
version.workspace = true

[dependencies]
```

## Building

The build command
```bash
cargo run -- build crate -l path/to/docs_rs_workspace_package/my_lib
```
fails with
```bash
Error: Building documentation failed

Caused by:
    Building documentation failed

Caused by:
    invalid Cargo.toml syntax
```
which makes sense due to
```toml
version.workspace = true
```

## Fix
However when running the following sequence of commands
```bash
# Run this in the directory of docs_rs_workspace_package
cargo package -p my_lib
```
and then building again
```bash
# Run this from the docs.rs repo
cargo run -- build crate -l path/to/docs_rs_workspace_package/target/package/my_lib-0.1.0
```
then the build succeeds.

File: README.md (hunk 1 above)
Comment (jonaspleyer) [reply README.md:211]: I think the most important thing would be make it really obvious how exactly packages are built. I do not understand this either at the moment. Just running
```
cargo +nightly rustdoc -- with the flags you think you need
```
is not identical to the build process but obviously the local build specified in the README also is (currently) not.

**Comment by GuillaumeGomez** [comment]
There is a `docs` folder. Don't hesitate to add more entries into it.

**Comment by jonaspleyer** [comment]
I have followed your suggestions. Somehow I feel a bit stressed when writing these things in the `/docs` folder. Would it not make sense to group them more concisely and either include this information on docs.rs itself or possibly as a book?

**Review by jonaspleyer — COMMENTED** [review]
I have inserted a more detailed description of when this problem occurs and how to circumvent it in the `docs` folder. I hope that this is sufficiently detailed to give a clear explanation.

**Review by syphar — APPROVED** [review]

**Pull request merged**